azure-mgmt-resources/src/main/java/com/azure/management/AzureServiceClient.java

```diff
@@ -85,10 +92,31 @@ public String userAgent() {
 
     private SerializerAdapter serializerAdapter = new AzureJacksonAdapter();
 
+    private String sdkName;
+
     public SerializerAdapter getSerializerAdapter() {
         return this.serializerAdapter;
     }
 
+    public Context getContext() {
+        Context context = new Context("java.version", JAVA_VERSION);
+        if (!CoreUtils.isNullOrEmpty(OS_NAME)) {
+            context = context.addData("os.name", OS_NAME);
+        }
+        if (!CoreUtils.isNullOrEmpty(OS_VERSION)) {
+            context = context.addData("os.version", OS_VERSION);
+        }
+        if (sdkName == null) {
+            String packageName = this.getClass().getPackage().getName();
+            if (packageName.endsWith(".models")) {
+                sdkName = packageName.substring(0, packageName.length() - ".models".length());

```

Comment:
I am doing a bit of trick here.

`ComputeManagementClientImpl` from `com.azure.management.compute.models` would have Sdk-Name `com.azure.management.compute`